app/clients/smtp/aws.py

```diff
@@ -0,0 +1,207 @@
+        "TXT")
+
+    credentials = add_user(iam_client, name)
+    return credentials
+
+
+def smtp_get_user_key(name):
+    try:
+        iam_client = boto3.client('iam', region_name=current_app.config["AWS_REGION"])
+        return iam_client.list_access_keys(
+            UserName=name,
+        )["AccessKeyMetadata"][0]["AccessKeyId"]
+    except Exception as e:
+        raise str(e)
+     
+
+def smtp_remove(name):
+    try:
+        ses_client = boto3.client('ses', region_name=current_app.config["AWS_REGION"])
+        r53_client = boto3.client('route53', region_name=current_app.config["AWS_REGION"])
+        iam_client = boto3.client('iam', region_name=current_app.config["AWS_REGION"])
+
+        [domain,_] = name.split("-")
+
+        policies = iam_client.list_user_policies(
+            UserName=name,
+        )["PolicyNames"]
+
+        for policy in policies:
+            iam_client.delete_user_policy(
+                UserName=name,
+                PolicyName=policy
+            )
+
+        keys = iam_client.list_access_keys(
+            UserName=name,
+        )["AccessKeyMetadata"]
+
+        for key in keys:
+            iam_client.delete_access_key(
+                UserName=name,
+                AccessKeyId=key["AccessKeyId"]
+            )
+
+        iam_client.delete_user(UserName=name)
+        ses_client.delete_identity(Identity=domain)
+        records = r53_client.list_resource_record_sets(
+            HostedZoneId=current_app.config["AWS_ROUTE53_ZONE"],
+            StartRecordName=domain
+        )["ResourceRecordSets"]
+
+        for record in records:
+            delete_record(r53_client, record)
+        
+        return True
+
+    except Exception as e:
+        raise str(e)
+
+
+def create_domain_identity(client, name):
+    try:
+        return client.verify_domain_identity(
+            Domain=name
+        )['VerificationToken']
+    except Exception as e:
+        raise str(e)
+
+
+def get_dkim(client, name):
+    try:
+        return client.verify_domain_dkim(
+            Domain=name
+        )['DkimTokens']
+    except Exception as e:
+        raise str(e)
+
+
+def add_user(client, name):
+    try:
+        user_name = name + "-" + str(int(time.time()))
+        client.create_user(
+            Path='/notification-smtp/',
+            UserName=user_name,
+            Tags=[
+                {
+                    'Key': 'SMTP-USER',
+                    'Value': name
+                },
+            ]
+        )
+
+        client.put_user_policy(
+            PolicyDocument=generate_user_policy(name),
+            PolicyName="SP-" + user_name,
+            UserName=user_name
+        )
+
+        response = client.create_access_key(
+            UserName=user_name
+        )
+
+        credentials = {
+            "iam": user_name,
+            "domain": name,
+            "name": "email-smtp.us-east-1.amazonaws.com",
```

Comment:
Guessing this is likely a fixed string, but is it worth pulling in from env just for future proofing and isolation of config from code?